packages/ag-charts-enterprise/src/features/crosshair/crosshair.ts

```diff
@@ -162,103 +186,150 @@ export class Crosshair extends _ModuleSupport.BaseModuleInstance implements _Mod
         if (!this.enabled) {
             return;
         }
 
-        const { crosshairGroup, seriesRect, axisCtx } = this;
+        const { crosshairGroup, axisCtx } = this;
         const { datum, series } = event.currentHighlight ?? {};
         const hasCrosshair = datum && (series?.axes.x?.id === axisCtx.axisId || series?.axes.y?.id === axisCtx.axisId);
 
         this.activeHighlight = hasCrosshair ? event.currentHighlight : undefined;
 
         if (this.snap) {
             if (!this.visible || !this.activeHighlight) {
-                this.hideCrosshair();
+                this.hideCrosshairs();
                 return;
             }
 
-            const { value, position } = this.getActiveHighlight(this.activeHighlight);
+            const activeHighlightData = this.getActiveHighlightData(this.activeHighlight);
+
+            this.updatePositions(activeHighlightData);
 
             crosshairGroup.visible = true;
+        }
+    }
+
+    private updatePositions(data: { [key: string]: { value: any; position: number } }) {
+        const { seriesRect, lineGroupSelection } = this;
+        lineGroupSelection.each((line, key) => {
+            const lineData = data[key];
+            if (!lineData) {
+                line.visible = false;
+                return;
+            }
+            line.visible = true;
 
+            const { value, position } = lineData;
             let x = 0;
             let y = 0;
-            if (axisCtx.direction === 'x') {
+            if (this.isVertical()) {
                 x = position;
-                crosshairGroup.translationX = Math.round(x + seriesRect.x);
+                line.translationX = Math.round(x);
             } else {
                 y = position;
-                crosshairGroup.translationY = Math.round(y + seriesRect.y);
+                line.translationY = Math.round(y);
             }
 
             if (this.label.enabled) {
-                this.showLabel(x + seriesRect.x, y + seriesRect.y, value);
+                this.showLabel(x + seriesRect.x, y + seriesRect.y, value, key);
             } else {
-                this.hideLabel();
+                this.hideLabel(key);
             }
+        });
+    }
+
+    private getData(event: _ModuleSupport.InteractionEvent<'hover'>): {
+        [key: string]: { position: number; value: any };
+    } {
+        const { seriesRect, axisCtx } = this;
+        const key = 'pointer';
+        const { datum, xKey = '' } = this.activeHighlight ?? {};
+        const { offsetX, offsetY } = event;
+
+        if (this.isVertical()) {
+            const x = offsetX - seriesRect.x;
+            return {
+                [key]: {
+                    position: x,
+                    value: axisCtx.continuous ? axisCtx.scaleInvert(x) : datum?.[xKey] ?? '',
+                },
+            };
+        } else {
+            const y = offsetY - seriesRect.y;
+            return {
```

Comment:
Maybe refactor to a single return point, given only the `position` and `xKey` vs `yKey` vary between these two control-flows?